pkg/server/backend_manager.go

```diff
@@ -125,25 +130,53 @@ func (b *backend) Context() context.Context {
 	return b.conn.Context()
 }
 
-func (b *backend) GetAgentIdentifiers() (header.Identifiers, error) {
+func (b *backend) GetAgentID() string {
+	return b.id
+}
+
+func (b *backend) GetAgentIdentifiers() header.Identifiers {
+	return b.idents
+}
+
+func getAgentID(stream agent.AgentService_ConnectServer) (string, error) {
+	md, ok := metadata.FromIncomingContext(stream.Context())
+	if !ok {
+		return "", fmt.Errorf("failed to get context")
+	}
+	agentIDs := md.Get(header.AgentID)
+	if len(agentIDs) != 1 {
+		return "", fmt.Errorf("expected one agent ID in the context, got %v", agentIDs)
+	}
+	return agentIDs[0], nil
+}
+
+func getAgentIdentifiers(conn agent.AgentService_ConnectServer) (header.Identifiers, error) {
 	var agentIdentifiers header.Identifiers
-	md, ok := metadata.FromIncomingContext(b.Context())
+	md, ok := metadata.FromIncomingContext(conn.Context())
 	if !ok {
 		return agentIdentifiers, fmt.Errorf("failed to get metadata from context")
```

Comment:
One subtle difference. In the old code, for the DefaultBackendManager/default strategy case, we would not call GetAgentIdentifiers. Now we call if for all strategies. So previously if there would have been an error getting agent identifiers while only using the default strategy, I believe it would have "worked" but now it won't. I think thats probably ok but its worth paying attention to.

Reply:
I missed that, good point.

Maybe there is a case for `konnectivity-server` gracefully handling bad identifiers: if the agent binary ever introduces a new identifier type, then with binary skew (new agent, old server) the server could have problems. 

If there is agreement I can update this PR.